file/writable_file_writer.h

```diff
@@ -188,6 +193,14 @@ class WritableFileWriter {
           file_checksum_gen_factory->CreateFileChecksumGenerator(
               checksum_gen_context);
     }
+
```

Comment:
This is bad as it silently drops the fact that the user requested data validation.  Why can't this be done at a higher level during options validation?